### PR TITLE
ENG-2425 UI updates to allow searching for hotels

### PR DESCRIPTION
**Pull request by fjlanasa**
JIRA: [ENG-2425](https://etmainc.atlassian.net/browse/ENG-2425)

Spec: https://docs.google.com/document/d/17OeDXXvgXWmlUvjn5MjACFnFUyLkvMp8Ba7ogr-x9iI/edit

Summary: Needed to update autocomplete module to allow for hotel results. Now, instead of creating only a `DestinationList` on receiving data from api, create a `Results` model, which itself has a `LocationList` and `PartnerPropertyList`, with logic determining how many to show of which. Also updated Search module to allow switching search location to nearest location of selected Partner Property. Currently, API is returning Boston's webID for every partner property, but will be updated this sprint. 

Related PRs:

Moments: https://github.com/ETMAInc/TruffleHunt-MarriottMoments/pull/814

### CODE REVIEW

File: src/autocomplete/results/models/index.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    d = { places: [], properties: [] },
+    maxDestinationResults = DEFAULT_MAX_DESTINATION_COUNT,
+    maxPropertyResults = DEFAULT_MAX_PROPERTY_COUNT
+  ) {
+    this.maxDestinationResults = maxDestinationResults;
+    this.maxPropertyResults = maxPropertyResults;
+    this.destinationList = this.initializeDestinationList(d.places);
+    this.propertiesList = this.initializePropertiesList(d.properties);
+  }
+
```
Comment (fjlanasa) [inline src/autocomplete/results/models/index.ts]: rename

File: src/autocomplete/destination_autocomplete.tsx [hunk 2]
```diff
@@ -2,15 +2,21 @@ import * as React from "react";
...
+import ResultsComponent from "./results/components";
 import Tools from "../utils/tools";
 import { FilterSection } from "../search/search";
 
 export interface IDestinationAutocompleteProps {
   apiUtils: IApiUtils;
   count: number;
-  selectedDestination?: any | Location | null;
+  maxDestinationResults?: number;
+  maxPropertyResults?: number;
```
Comment (fjlanasa) [inline src/autocomplete/destination_autocomplete.tsx]: remove any as type

File: src/autocomplete/destination_autocomplete.tsx [hunk 3]
```diff
@@ -57,9 +63,13 @@ export default class DestinationAutocomplete extends React.Component<
     };
   }
 
-  initSelectedDestination(destination): Location {
-    if (destination) {
+  initSelectedDestination(destination): null | Location | PartnerProperty {
+    if (destination && (destination.WebId || destination.Id)) {
       return new Location(destination);
```
Comment (fjlanasa) [inline src/autocomplete/destination_autocomplete.tsx]: explore removing

File: src/search/search.tsx [hunk 4]
```diff
@@ -117,22 +126,22 @@ export default class Search extends React.Component<
     };
   }
```
Comment (fjlanasa) [inline src/search/search.tsx]: clean up naming

File: src/autocomplete/destination_autocomplete.tsx [hunk 5]
```diff
@@ -234,22 +239,47 @@ export default class DestinationAutocomplete extends React.Component<
...
+        window.location.href = destinationPath;
       } else {
         window.location.href = searchPath;
       }
     }
   }
 
+  navigateToPartnerPropertyUrl() {
+    let { selectedDestination } = this.state;
+    if (selectedDestination instanceof PartnerProperty) {
```
Comment (brianeroth) [inline src/autocomplete/destination_autocomplete.tsx]: @fjlanasa Let's talk about using `instanceof` and alternatives when you get back

**Pull request merged**